### ISSUE TITLE
Storage permission not needed

### ISSUE THREAD

**Issue by Blaisorblade** [farkam135/GoIV#419]
We ask write permissions to write to `getExternalFilesDir(null)`, but according to its JavaDocs since Android KitKat (API 19) we don't need this permission.

**Comment by DJCrapsody** [comment]
Reading app specific directories do not require read or write external storage permissions. Unless we are reading from other external directories (screenshot mode?).

**Comment by Blaisorblade** [comment]
Right, thanks! We're reading media storage in screenshot mode (which requires `READ_EXTERNAL_STORAGE`, http://stackoverflow.com/a/30626431/53974), and writing there optionally to delete the screenshot.

Both are classified as dangerous permissions.
However, this permission is not needed in non-screenshot mode.

**Comment by sarav** [comment]
Ok, but what fix do we need for this? Battery saver mode is support in all versions of Android. So, we'll just have to live with this?

**Comment by Blaisorblade** [comment]
We already use Android 6 runtime permissions where available. But we defeat the point by asking for all permissions at the start, whether we need them or not. I think that was because of the OCR data—I remember reading this before I started looking at the code.

**Comment by DJCrapsody** [comment]
ocr data don't require the permission as it writes into externalFilesDir so the only time required for storage permissions is when we use screenshot mode.

And this probably wouldn't affect  ~Lollipop devices right?

**Comment by Blaisorblade** [comment]
@DJCrapsody IIUC yes to both.
